src/scripts/main-nav.js

```diff
@@ -0,0 +1,235 @@
+    signupButton.parentNode.style.fontSize = '0';
+  }
+
+  function makeLogoutCall (e) {
+
+    e.preventDefault();
+    e.stopPropagation();
+    var logoutRequest = new XMLHttpRequest();
+    logoutRequest.open('POST', '/api/developer/sign_out', true);
+    logoutRequest.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded; charset=UTF-8');
+    logoutRequest.send();
+
+
+    // change Button UI, cancle the event when log out is successful
+
+    logoutRequest.onload = function () {
+      if (isThisDevPortalPage() || isThisCustomExtractPage()) {
+        // If user was on dashboard or custom extract page, redirect the page to sign in page
+        window.location = '/developers/sign_in/'
+      } else {
+        // If user was on other pages than dashboard and custom extract page, just refresh the page.
+        window.location.reload();
+      }
+    }
+
+    logoutRequest.onerror = function () {
+
+    }
+  }
+
+  function getLoginElem (nickname, avatarImageURL) {
+    var strVar = '';
+    strVar += '<a id="sign-in" class="dropdown-toggle" data-toggle="dropdown" data-target="#" role="button">';
+    strVar += ' <div id=\"login-profile\">';
+    strVar += '   <img width=\"18\" height=\"18\" src=\"'+avatarImageURL+'\" style=\"border-radius: 50%; margin-top: -9px;\">';
+    strVar += ' <\/div>';
+    strVar += ' <div class="login-txt"> ' + nickname + ' <\/div>';
+    strVar += '</a>';
+    strVar += ' <ul class="dropdown-menu">';
+    strVar += '   <li><a href="/developers/">Dashboard</a></li>';
+    strVar += '   <li><a href="/data/metro-extracts/your-extracts/">Custom Extracts</a></li>';
+    strVar += '   <li id="sign-out"><a href="#"> Logout</a></li>';
+    strVar += ' </ul>';
+    return strVar;
+  }
+
+  function getNotLoginElem () {
+    var strVar='';
+    // strVar += '<a id=\"login\" href=\"\/developers\/sign_in\">';
+    strVar += '  <div id=\"login-profile\">';
+    strVar += '    <div class=\"compass\">';
+    strVar += '      <div class=\"center-dot\"><\/div>';
+    strVar += '    <\/div>';
+    strVar += '  <\/div>';
+    strVar += '  <div class=\"login-txt\">sign in<\/div>';
+    // strVar += '<\/a>';
+    return strVar;
+  }
+
+  function getSignUpElem () {
+    var strVar = '';
+    strVar += '<span class="visible-xs-inline visible-sm-inline visible-md-inline visible-lg-inline btn btn-mapzen">';
+    strVar += '  sign up';
+    strVar += '<\/span>';
+    // strVar += '<div class="alert alert-success">';
+    // strVar += ' <strong>Success!</strong> Indicates a successful or positive action.';
+    // strVar += '</div>';
+    return strVar;
+  }
+
+  function getSignOutElem () {
+    return 'sign out';
+  }
+
+  function putActiveTab () {
+    var navItems = document.querySelectorAll('.navbar-nav>li');
+    // After dom manipulation
+    var loginButton = document.getElementById('sign-in');
+    var signupButton = document.getElementById('sign-up');
+
+    if (isThisDevPortalPage()) {
```

Comment:
if we're still using the old html, when we get here loginButton and signupButton will still be null, which will cause `loginButton.parentNode` to throw `Uncaught TypeError: Cannot read property 'parentNode' of null`.
can we update to `loginButton && isThisDevPortalPage()` here? or something equivalent?